hex_blueprints/check_run_status.py

```diff
@@ -0,0 +1,128 @@
+from dataclasses import dataclass
+import requests
+import shipyard_utils as shipyard
+import argparse
+import sys
+import re
+
+try:
+    import exit_codes as ec
+except BaseException:
+    from . import exit_codes as ec
+@dataclass
+class HexResponse:
+    """
+    This is a utility class to maintain the http status code with the json response
+    """
+    status_code : int
+    response_json : dict
+    
+def get_args():    
+    """
+    Creates the argument parser for the CLI
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--project-id",dest='project_id',required=True)
+    parser.add_argument('--api-token',dest = 'api_token',required=True)
+    parser.add_argument('--run-id',dest = 'run_id',required=False)
+    ## add a flag for input params 
+    args = parser.parse_args()
+    return args
+
+def get_run_status(project_id, api_token, run_id):
+    """
+    Returns the json with metadata of the last run of the project
+    """
+    pattern = re.compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-4[0-9a-fA-F]{3}-[89ABab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$") ## copied from Hex's error message
```

Comment:
Can you link to Hex's error message that you're trying to match? This looks like a modified UUID.

Reply:
It probably doesn't need to change at the moment, but this pattern is _more_ restrictive than the UUID v4 allows.